src/convert/impls.rs

```diff
@@ -624,3 +624,163 @@ where
     }
     result.into_boxed_slice()
 }
+
+// WebAssembly ToWebAssemblyValue implementations for primitive types
+// These implement the exact WebAssembly specification algorithms
+
+impl TryFromJsValue for i32 {
```

Comment:
I think it would be nice to split those changes out into a separate PR, makes a bit confusing to see what is related to async import support specifically and what are these traits for.